pkg/hnsw/hnsw.go

```diff
@@ -127,3 +126,23 @@ func (h *Hnsw) searchLayer(q Vector, ef, layerId int, nearestNeighborsToQForEf *
 		}
 	}
 }
+
+func (h *Hnsw) selectNeighbors(candidates *MinQueue, numNeighborsToReturn int) *MinQueue {
+
+	if candidates.Len() <= numNeighborsToReturn {
+		return nil
+	}
+
+	mCandidatesNearestElementsFromQ := NewMinQueue()
+
+	for !mCandidatesNearestElementsFromQ.IsEmpty() {
```

Comment:
```suggestion
	for candidate := candidates.Peel(); candidate != nil; candidate = candidates.Peel() {
```

could also work, up to you.

Reply:
Done.